diff --git a/browser/amethyst/content/browser.js b/browser/amethyst/content/browser.js
new file mode 100644
--- /dev/null
+++ b/browser/amethyst/content/browser.js
@@ -0,0 +1,22 @@
+function onload(){
+  window.fullScreen = true;
+
+  let uriToLoad = "about:blank";
+  if (window.arguments && window.arguments.length) {
+    let args = window.arguments[0];
+    if (args instanceof Ci.nsIArray && args.length) {
+      uriToLoad = args.queryElementAt(0, Ci.nsISupportsString).data;
+    }
+  }
+
+  this.content = document.getElementById("content");
+  this.content.src = uriToLoad;
+}
+
+function BrowserBack(){
+  return this.content.goBack();
+}
+
+function BrowserFullScreen(){
+  window.fullScreen = !window.fullScreen;
+}
diff --git a/browser/amethyst/content/browser.xul b/browser/amethyst/content/browser.xul
new file mode 100644
--- /dev/null
+++ b/browser/amethyst/content/browser.xul
@@ -0,0 +1,32 @@
+#filter substitution
+<?xml version="1.0"?>
+<!DOCTYPE window [
+<!ENTITY % brandDTD SYSTEM "chrome://branding/locale/brand.dtd" >
+%brandDTD;
+]>
+
+<window id="main-window"
+        xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"
+        width="800"
+        height="480"
+        sizemode="maximized"
+        title="&brandFullName;"
+        onload="onload()"
+>
+
+<commandset id="mainCommandSet">
+  <command id="Browser:Back"    oncommand="BrowserBack();"/>
+  <command id="View:FullScreen" oncommand="BrowserFullScreen();"/>
+</commandset>
+<keyset id="mainKeyset">
+  <key keycode="VK_BACK" command="Browser:Back"/>
+  <key id="key_fullScreen" keycode="VK_F11" command="View:FullScreen"/>
+</keyset>
+
+<script type="application/javascript">
+  Components.utils.import("resource://gre/modules/Services.jsm");
+  Services.scriptloader.loadSubScript("chrome://browser/content/browser.js", this);
+</script>
+
+<browser id="content" type="content" flex="1" />
+</window>
diff --git a/browser/amethyst/jar.mn b/browser/amethyst/jar.mn
new file mode 100644
--- /dev/null
+++ b/browser/amethyst/jar.mn
@@ -0,0 +1,9 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+browser.jar:
+%  content browser %content/browser/ contentaccessible=yes
+
+#*       content/browser/browser.css                   (content/browser.css)
+        content/browser/browser.js                    (content/browser.js)
+*       content/browser/browser.xul                   (content/browser.xul)
diff --git a/browser/amethyst/moz.build b/browser/amethyst/moz.build
new file mode 100644
--- /dev/null
+++ b/browser/amethyst/moz.build
@@ -0,0 +1,7 @@
+# -*- Mode: python; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+JAR_MANIFESTS += ['jar.mn']
diff --git a/browser/base/jar.mn b/browser/base/jar.mn
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -44,9 +44,6 @@ browser.jar:
         content/browser/aboutTabCrashed.css           (content/aboutTabCrashed.css)
         content/browser/aboutTabCrashed.js            (content/aboutTabCrashed.js)
         content/browser/aboutTabCrashed.xhtml         (content/aboutTabCrashed.xhtml)
-*       content/browser/browser.css                   (content/browser.css)
-        content/browser/browser.js                    (content/browser.js)
-*       content/browser/browser.xul                   (content/browser.xul)
         content/browser/browser-addons.js             (content/browser-addons.js)
         content/browser/browser-captivePortal.js      (content/browser-captivePortal.js)
         content/browser/browser-ctrlTab.js            (content/browser-ctrlTab.js)
diff --git a/browser/moz.build b/browser/moz.build
--- a/browser/moz.build
+++ b/browser/moz.build
@@ -12,6 +12,7 @@ with Files('docs/**'):
     SCHEDULES.exclusive = ['docs']
 
 DIRS += [
+    'amethyst',
     'base',
     'components',
     'experiments',
