diff --git a/browser/actors/ContextMenuChild.jsm b/browser/actors/ContextMenuChild.jsm
--- a/browser/actors/ContextMenuChild.jsm
+++ b/browser/actors/ContextMenuChild.jsm
@@ -133,21 +133,6 @@ class ContextMenuChild extends JSWindowA
                   media.requestFullscreen();
                 }
                 break;
-              case "pictureinpicture":
-                Services.telemetry.keyedScalarAdd(
-                  "pictureinpicture.opened_method",
-                  "contextmenu",
-                  1
-                );
-                let event = new this.contentWindow.CustomEvent(
-                  "MozTogglePictureInPicture",
-                  {
-                    bubbles: true,
-                  },
-                  this.contentWindow
-                );
-                media.dispatchEvent(event);
-                break;
             }
           }
         );
diff --git a/browser/base/content/browser-context.inc b/browser/base/content/browser-context.inc
--- a/browser/base/content/browser-context.inc
+++ b/browser/base/content/browser-context.inc
@@ -182,10 +182,6 @@
       <menuitem id="context-viewvideo"
                 data-l10n-id="main-context-menu-video-view-new-tab"
                 oncommand="gContextMenu.viewMedia(event);"/>
-      <menuitem id="context-video-pictureinpicture"
-                data-l10n-id="main-context-menu-media-watch-pip"
-                type="checkbox"
-                oncommand="gContextMenu.mediaCommand('pictureinpicture');"/>
       <menuseparator id="context-media-sep-commands"/>
       <menuitem id="context-reloadimage"
                 data-l10n-id="main-context-menu-image-reload"
diff --git a/browser/base/content/browser-sets.inc b/browser/base/content/browser-sets.inc
--- a/browser/base/content/browser-sets.inc
+++ b/browser/base/content/browser-sets.inc
@@ -43,7 +43,6 @@
     <command id="View:PageInfo" oncommand="BrowserPageInfo();"/>
     <command id="View:FullScreen" oncommand="BrowserFullScreen();"/>
     <command id="View:ReaderView" oncommand="AboutReaderParent.toggleReaderMode(event);"/>
-    <command id="View:PictureInPicture" oncommand="PictureInPicture.onCommand(event);"/>
     <command id="cmd_find" oncommand="gLazyFindCommand('onFindCommand')"/>
     <command id="cmd_findAgain" oncommand="gLazyFindCommand('onFindAgainCommand', false)"/>
     <command id="cmd_findPrevious" oncommand="gLazyFindCommand('onFindAgainCommand', true)"/>
@@ -234,14 +233,6 @@
 #endif
          disabled="true"/>
 
-#ifndef XP_MACOSX
-    <key id="key_togglePictureInPicture" data-l10n-id="picture-in-picture-toggle-shortcut" command="View:PictureInPicture" modifiers="accel,shift"/>
-    <key                                 data-l10n-id="picture-in-picture-toggle-shortcut-alt" command="View:PictureInPicture" modifiers="accel,shift"/>
-#else
-    <key                                 data-l10n-id="picture-in-picture-toggle-shortcut-mac" command="View:PictureInPicture" modifiers="accel,alt,shift"/>
-    <key                                 data-l10n-id="picture-in-picture-toggle-shortcut-mac-alt" command="View:PictureInPicture" modifiers="accel,alt,shift"/>
-#endif
-
     <key data-l10n-id="nav-reload-shortcut" command="Browser:Reload" modifiers="accel" id="key_reload"/>
     <key data-l10n-id="nav-reload-shortcut" command="Browser:ReloadSkipCache" modifiers="accel,shift" id="key_reload_skip_cache"/>
     <key id="key_viewSource" data-l10n-id="page-source-shortcut" command="View:PageSource" modifiers="accel"/>
diff --git a/browser/base/content/main-popupset.inc.xhtml b/browser/base/content/main-popupset.inc.xhtml
--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -423,12 +423,6 @@
 #include browser-context.inc
   </menupopup>
 
-  <menupopup id="pictureInPictureToggleContextMenu">
-    <menuitem label="&pictureInPictureHideToggle.label;"
-              accesskey="&pictureInPictureHideToggle.accesskey;"
-              oncommand="PictureInPicture.hideToggle();" />
-  </menupopup>
-
 #include ../../components/places/content/placesContextMenu.inc.xhtml
 
   <panel id="ctrlTab-panel" hidden="true" norestorefocus="true" level="top" orient="vertical">
diff --git a/browser/base/content/nsContextMenu.js b/browser/base/content/nsContextMenu.js
--- a/browser/base/content/nsContextMenu.js
+++ b/browser/base/content/nsContextMenu.js
@@ -872,16 +872,6 @@ class nsContextMenu {
       "context-video-fullscreen",
       this.onVideo && !this.target.ownerDocument.fullscreen
     );
-    {
-      let shouldDisplay =
-        Services.prefs.getBoolPref(
-          "media.videocontrols.picture-in-picture.enabled"
-        ) &&
-        this.onVideo &&
-        !this.target.ownerDocument.fullscreen &&
-        this.target.readyState > 0;
-      this.showItem("context-video-pictureinpicture", shouldDisplay);
-    }
     this.showItem("context-media-eme-learnmore", this.onDRMMedia);
 
     // Disable them when there isn't a valid media source loaded.
diff --git a/toolkit/actors/UAWidgetsChild.jsm b/toolkit/actors/UAWidgetsChild.jsm
--- a/toolkit/actors/UAWidgetsChild.jsm
+++ b/toolkit/actors/UAWidgetsChild.jsm
@@ -88,11 +88,6 @@ class UAWidgetsChild extends JSWindowAct
         uri = "chrome://global/content/elements/videocontrols.js";
         widgetName = "VideoControlsWidget";
         prefKeys = [
-          "media.videocontrols.picture-in-picture.video-toggle.enabled",
-          "media.videocontrols.picture-in-picture.video-toggle.always-show",
-          "media.videocontrols.picture-in-picture.video-toggle.min-video-secs",
-          "media.videocontrols.picture-in-picture.video-toggle.position",
-          "media.videocontrols.picture-in-picture.video-toggle.has-used",
           "media.videocontrols.keyboard-tab-to-all-controls",
         ];
         break;
diff --git a/toolkit/actors/moz.build b/toolkit/actors/moz.build
--- a/toolkit/actors/moz.build
+++ b/toolkit/actors/moz.build
@@ -48,7 +48,6 @@ FINAL_TARGET_FILES.actors += [
     "InlineSpellCheckerChild.jsm",
     "InlineSpellCheckerParent.jsm",
     "KeyPressEventModelCheckerChild.jsm",
-    "PictureInPictureChild.jsm",
     "PopupBlockingChild.jsm",
     "PopupBlockingParent.jsm",
     "PrintingChild.jsm",
diff --git a/toolkit/components/moz.build b/toolkit/components/moz.build
--- a/toolkit/components/moz.build
+++ b/toolkit/components/moz.build
@@ -56,7 +56,6 @@ DIRS += [
     "passwordmgr",
     "pdfjs",
     "perfmonitoring",
-    "pictureinpicture",
     "places",
     "processtools",
     "processsingleton",
diff --git a/toolkit/content/widgets/videocontrols.js b/toolkit/content/widgets/videocontrols.js
--- a/toolkit/content/widgets/videocontrols.js
+++ b/toolkit/content/widgets/videocontrols.js
@@ -51,8 +51,6 @@ this.VideoControlsWidget = class {
       newImpl = VideoControlsImplWidget;
     } else if (this.isMobile) {
       newImpl = NoControlsMobileImplWidget;
-    } else if (VideoControlsWidget.isPictureInPictureVideo(this.element)) {
-      newImpl = NoControlsPictureInPictureImplWidget;
     } else if (
       pageURI.startsWith("http://") ||
       pageURI.startsWith("https://")
@@ -66,8 +64,7 @@ this.VideoControlsWidget = class {
     // without a value change.
     if (
       this.impl &&
-      this.impl.constructor == newImpl &&
-      this.impl.elementStateMatches(this.element)
+      this.impl.constructor == newImpl
     ) {
       return;
     }
@@ -109,78 +106,6 @@ this.VideoControlsWidget = class {
     this.impl.onPrefChange(prefName, prefValue);
   }
 
-  static isPictureInPictureVideo(someVideo) {
-    return someVideo.isCloningElementVisually;
-  }
-
-  /**
-   * Returns true if a <video> meets the requirements to show the Picture-in-Picture
-   * toggle. Those requirements currently are:
-   *
-   * 1. The video must be 45 seconds in length or longer.
-   * 2. Neither the width or the height of the video can be less than 140px.
-   * 3. The video must have audio.
-   * 4. The video must not a MediaStream video (Bug 1592539)
-   *
-   * This can be overridden via the
-   * media.videocontrols.picture-in-picture.video-toggle.always-show pref, which
-   * is mostly used for testing.
-   *
-   * @param {Object} prefs
-   *   The preferences set that was passed to the UAWidget.
-   * @param {Element} someVideo
-   *   The <video> to test.
-   * @param {Object} reflowedDimensions
-   *   An object representing the reflowed dimensions of the <video>. Properties
-   *   are:
-   *
-   *     videoWidth (Number):
-   *       The width of the video in pixels.
-   *
-   *     videoHeight (Number):
-   *       The height of the video in pixels.
-   *
-   * @return {Boolean}
-   */
-  static shouldShowPictureInPictureToggle(
-    prefs,
-    someVideo,
-    reflowedDimensions
-  ) {
-    if (isNaN(someVideo.duration)) {
-      return false;
-    }
-
-    if (
-      prefs["media.videocontrols.picture-in-picture.video-toggle.always-show"]
-    ) {
-      return true;
-    }
-
-    const MIN_VIDEO_LENGTH =
-      prefs[
-        "media.videocontrols.picture-in-picture.video-toggle.min-video-secs"
-      ];
-
-    if (someVideo.duration < MIN_VIDEO_LENGTH) {
-      return false;
-    }
-
-    const MIN_VIDEO_DIMENSION = 140; // pixels
-    if (
-      reflowedDimensions.videoWidth < MIN_VIDEO_DIMENSION ||
-      reflowedDimensions.videoHeight < MIN_VIDEO_DIMENSION
-    ) {
-      return false;
-    }
-
-    if (!someVideo.mozHasAudio) {
-      return false;
-    }
-
-    return true;
-  }
-
   /**
    * Some variations on the Picture-in-Picture toggle are being experimented with.
    * These variations have slightly different setup parameters from the currently
@@ -260,7 +185,6 @@ this.VideoControlsImplWidget = class {
       controlsOverlay: null,
       fullscreenButton: null,
       layoutControls: null,
-      isShowingPictureInPictureMessage: false,
 
       textTracksCount: 0,
       videoEvents: [
@@ -301,7 +225,6 @@ this.VideoControlsImplWidget = class {
       set isAudioOnly(val) {
         this._isAudioOnly = val;
         this.setFullscreenButtonState();
-        this.updatePictureInPictureToggleDisplay();
 
         if (!this.isTopLevelSyntheticDocument) {
           return;
@@ -451,17 +374,6 @@ this.VideoControlsImplWidget = class {
           this.statusIcon.setAttribute("type", "error");
           this.updateErrorText();
           this.setupStatusFader(true);
-        } else if (VideoControlsWidget.isPictureInPictureVideo(this.video)) {
-          this.setShowPictureInPictureMessage(true);
-        }
-
-        if (this.video.readyState >= this.video.HAVE_METADATA) {
-          // According to the spec[1], at the HAVE_METADATA (or later) state, we know
-          // the video duration and dimensions, which means we can calculate whether or
-          // not to show the Picture-in-Picture toggle now.
-          //
-          // [1]: https://www.w3.org/TR/html50/embedded-content-0.html#dom-media-have_metadata
-          this.updatePictureInPictureToggleDisplay();
         }
 
         let adjustableControls = [
@@ -574,32 +486,6 @@ this.VideoControlsImplWidget = class {
         this.updateVolumeControls();
       },
 
-      updatePictureInPictureToggleDisplay() {
-        if (this.isAudioOnly) {
-          this.pictureInPictureToggle.hidden = true;
-          return;
-        }
-
-        if (
-          this.pipToggleEnabled &&
-          !this.isShowingPictureInPictureMessage &&
-          VideoControlsWidget.shouldShowPictureInPictureToggle(
-            this.prefs,
-            this.video,
-            this.reflowedDimensions
-          )
-        ) {
-          this.pictureInPictureToggle.hidden = false;
-          VideoControlsWidget.setupToggle(
-            this.prefs,
-            this.pictureInPictureToggle,
-            this.reflowedDimensions
-          );
-        } else {
-          this.pictureInPictureToggle.hidden = true;
-        }
-      },
-
       setupNewLoadState() {
         // For videos with |autoplay| set, we'll leave the controls initially hidden,
         // so that they don't get in the way of the playing video. Otherwise we'll
@@ -611,8 +497,7 @@ this.VideoControlsImplWidget = class {
           shouldShow &&
           !this.isAudioOnly &&
           this.video.currentTime == 0 &&
-          !this.hasError() &&
-          !this.isShowingPictureInPictureMessage;
+          !this.hasError();
         this.startFade(this.clickToPlay, shouldClickToPlayShow, true);
         this.startFade(this.controlBar, shouldShow, true);
       },
@@ -767,10 +652,8 @@ this.VideoControlsImplWidget = class {
               this.muteButton.disabled = true;
             }
             this.adjustControlSize();
-            this.updatePictureInPictureToggleDisplay();
             break;
           case "durationchange":
-            this.updatePictureInPictureToggleDisplay();
             break;
           case "loadeddata":
             this.firstFrameShown = true;
@@ -925,7 +808,6 @@ this.VideoControlsImplWidget = class {
             this.updateReflowedDimensions();
             this.reflowTriggeringCallValidator.isReflowTriggeringPropsAllowed = false;
             this.adjustControlSize();
-            this.updatePictureInPictureToggleDisplay();
             break;
           case "fullscreenchange":
             this.onFullscreenChange();
@@ -1069,11 +951,6 @@ this.VideoControlsImplWidget = class {
         );
       },
 
-      setShowPictureInPictureMessage(showMessage) {
-        this.pictureInPictureOverlay.hidden = !showMessage;
-        this.isShowingPictureInPictureMessage = showMessage;
-      },
-
       hasSources() {
         if (
           this.video.hasAttribute("src") &&
@@ -2462,9 +2339,6 @@ this.VideoControlsImplWidget = class {
         this.controlsOverlay = this.shadowRoot.getElementById(
           "controlsOverlay"
         );
-        this.pictureInPictureOverlay = this.shadowRoot.getElementById(
-          "pictureInPictureOverlay"
-        );
         this.controlsSpacer = this.shadowRoot.getElementById("controlsSpacer");
         this.clickToPlay = this.shadowRoot.getElementById("clickToPlay");
         this.fullscreenButton = this.shadowRoot.getElementById(
@@ -2478,9 +2352,6 @@ this.VideoControlsImplWidget = class {
         this.textTrackListContainer = this.shadowRoot.getElementById(
           "textTrackListContainer"
         );
-        this.pictureInPictureToggle = this.shadowRoot.getElementById(
-          "pictureInPictureToggle"
-        );
 
         if (this.positionDurationBox) {
           this.durationSpan = this.positionDurationBox.getElementsByTagName(
@@ -2778,31 +2649,12 @@ this.VideoControlsImplWidget = class {
             <bdi class="statusLabel" id="errorGeneric">&error.generic;</bdi>
           </div>
 
-          <div id="pictureInPictureOverlay" class="pictureInPictureOverlay stackItem" status="pictureInPicture" hidden="true">
-            <div class="statusIcon" type="pictureInPicture"></div>
-            <bdi class="statusLabel" id="pictureInPicture">&status.pictureInPicture;</bdi>
-          </div>
-
           <div id="controlsOverlay" class="controlsOverlay stackItem" role="none">
             <div class="controlsSpacerStack">
               <div id="controlsSpacer" class="controlsSpacer stackItem" role="none"></div>
               <button id="clickToPlay" class="clickToPlay" hidden="true"></button>
             </div>
 
-            <button id="pictureInPictureToggle" class="pip-wrapper" position="left" hidden="true">
-              <div class="pip-small clickable"></div>
-              <div class="pip-expanded clickable">
-                <span class="pip-icon-label clickable">
-                  <span class="pip-icon"></span>
-                  <span class="pip-label">&pictureInPictureToggle.label;</span>
-                </span>
-                <div class="pip-explainer clickable">
-                  &pictureInPictureExplainer;
-                </div>
-              </div>
-              <div class="pip-icon clickable"></div>
-            </button>
-
             <div id="controlBar" class="controlBar" role="none" hidden="true">
               <button id="playButton"
                       class="button playButton"
@@ -2873,11 +2725,6 @@ this.VideoControlsImplWidget = class {
     );
   }
 
-  elementStateMatches(element) {
-    let elementInPiP = VideoControlsWidget.isPictureInPictureVideo(element);
-    return this.isShowingPictureInPictureMessage == elementInPiP;
-  }
-
   destructor() {
     this.Utils.terminate();
     this.TouchUtils.terminate();
@@ -2888,7 +2735,6 @@ this.VideoControlsImplWidget = class {
 
   onPrefChange(prefName, prefValue) {
     this.prefs[prefName] = prefValue;
-    this.Utils.updatePictureInPictureToggleDisplay();
   }
 
   _setupEventListeners() {
@@ -3041,10 +2887,6 @@ this.NoControlsMobileImplWidget = class 
     );
   }
 
-  elementStateMatches(element) {
-    return true;
-  }
-
   destructor() {
     this.Utils.terminate();
   }
@@ -3085,63 +2927,6 @@ this.NoControlsMobileImplWidget = class 
   }
 };
 
-this.NoControlsPictureInPictureImplWidget = class {
-  constructor(shadowRoot, prefs) {
-    this.shadowRoot = shadowRoot;
-    this.prefs = prefs;
-    this.element = shadowRoot.host;
-    this.document = this.element.ownerDocument;
-    this.window = this.document.defaultView;
-  }
-
-  onsetup(direction) {
-    this.generateContent();
-
-    this.shadowRoot.firstElementChild.setAttribute("localedir", direction);
-  }
-
-  elementStateMatches(element) {
-    return true;
-  }
-
-  destructor() {}
-
-  onPrefChange(prefName, prefValue) {
-    this.prefs[prefName] = prefValue;
-  }
-
-  generateContent() {
-    /*
-     * Pass the markup through XML parser purely for the reason of loading the localization DTD.
-     * Remove it when migrate to Fluent.
-     */
-    const parser = new this.window.DOMParser();
-    parser.forceEnableDTD();
-    let parserDoc = parser.parseFromString(
-      `<!DOCTYPE bindings [
-      <!ENTITY % videocontrolsDTD SYSTEM "chrome://global/locale/videocontrols.dtd">
-      %videocontrolsDTD;
-      ]>
-      <div class="videocontrols" xmlns="http://www.w3.org/1999/xhtml" role="none">
-        <link rel="stylesheet" href="chrome://global/skin/media/videocontrols.css" />
-        <div id="controlsContainer" class="controlsContainer" role="none">
-          <div class="pictureInPictureOverlay stackItem" status="pictureInPicture">
-            <div id="statusIcon" class="statusIcon" type="pictureInPicture"></div>
-            <bdi class="statusLabel" id="pictureInPicture">&status.pictureInPicture;</bdi>
-          </div>
-          <div class="controlsOverlay stackItem"></div>
-        </div>
-      </div>`,
-      "application/xml"
-    );
-    this.shadowRoot.importNodeAndAppendChildAt(
-      this.shadowRoot,
-      parserDoc.documentElement,
-      true
-    );
-  }
-};
-
 this.NoControlsDesktopImplWidget = class {
   constructor(shadowRoot, prefs) {
     this.shadowRoot = shadowRoot;
@@ -3169,7 +2954,6 @@ this.NoControlsDesktopImplWidget = class
           }
           case "resizevideocontrols": {
             this.updateReflowedDimensions();
-            this.updatePictureInPictureToggleDisplay();
             break;
           }
           case "durationchange":
@@ -3177,32 +2961,11 @@ this.NoControlsDesktopImplWidget = class
           case "emptied":
           // Intentional fall-through
           case "loadedmetadata": {
-            this.updatePictureInPictureToggleDisplay();
             break;
           }
         }
       },
 
-      updatePictureInPictureToggleDisplay() {
-        if (
-          this.pipToggleEnabled &&
-          VideoControlsWidget.shouldShowPictureInPictureToggle(
-            this.prefs,
-            this.video,
-            this.reflowedDimensions
-          )
-        ) {
-          this.pictureInPictureToggle.hidden = false;
-          VideoControlsWidget.setupToggle(
-            this.prefs,
-            this.pictureInPictureToggle,
-            this.reflowedDimensions
-          );
-        } else {
-          this.pictureInPictureToggle.hidden = true;
-        }
-      },
-
       init(shadowRoot, prefs) {
         this.shadowRoot = shadowRoot;
         this.prefs = prefs;
@@ -3212,27 +2975,10 @@ this.NoControlsDesktopImplWidget = class
         this.window = this.document.defaultView;
         this.shadowRoot = shadowRoot;
 
-        this.pictureInPictureToggle = this.shadowRoot.getElementById(
-          "pictureInPictureToggle"
-        );
-
         if (this.document.fullscreenElement) {
           this.videocontrols.setAttribute("inDOMFullscreen", true);
         }
 
-        // Default the Picture-in-Picture toggle button to being hidden. We might unhide it
-        // later if we determine that this video is qualified to show it.
-        this.pictureInPictureToggle.hidden = true;
-
-        if (this.video.readyState >= this.video.HAVE_METADATA) {
-          // According to the spec[1], at the HAVE_METADATA (or later) state, we know
-          // the video duration and dimensions, which means we can calculate whether or
-          // not to show the Picture-in-Picture toggle now.
-          //
-          // [1]: https://www.w3.org/TR/html50/embedded-content-0.html#dom-media-have_metadata
-          this.updatePictureInPictureToggleDisplay();
-        }
-
         this.document.addEventListener("fullscreenchange", this, {
           capture: true,
         });
@@ -3277,17 +3023,12 @@ this.NoControlsDesktopImplWidget = class
     this.Utils.init(this.shadowRoot, this.prefs);
   }
 
-  elementStateMatches(element) {
-    return true;
-  }
-
   destructor() {
     this.Utils.terminate();
   }
 
   onPrefChange(prefName, prefValue) {
     this.prefs[prefName] = prefValue;
-    this.Utils.updatePictureInPictureToggleDisplay();
   }
 
   generateContent() {
@@ -3307,19 +3048,6 @@ this.NoControlsDesktopImplWidget = class
 
         <div id="controlsContainer" class="controlsContainer" role="none">
           <div class="controlsOverlay stackItem">
-            <button id="pictureInPictureToggle" class="pip-wrapper" position="left" hidden="true">
-              <div class="pip-small clickable"></div>
-              <div class="pip-expanded clickable">
-                <span class="pip-icon-label clickable">
-                  <span class="pip-icon"></span>
-                  <span class="pip-label">&pictureInPictureToggle.label;</span>
-                </span>
-                <div class="pip-explainer clickable">
-                  &pictureInPictureExplainer;
-                </div>
-              </div>
-              <div class="pip-icon"></div>
-            </button>
           </div>
         </div>
       </div>`,
diff --git a/toolkit/modules/ActorManagerParent.jsm b/toolkit/modules/ActorManagerParent.jsm
--- a/toolkit/modules/ActorManagerParent.jsm
+++ b/toolkit/modules/ActorManagerParent.jsm
@@ -302,46 +302,6 @@ let JSWINDOWACTORS = {
     },
   },
 
-  PictureInPictureLauncher: {
-    parent: {
-      moduleURI: "resource://gre/modules/PictureInPicture.jsm",
-    },
-    child: {
-      moduleURI: "resource://gre/actors/PictureInPictureChild.jsm",
-      events: {
-        MozTogglePictureInPicture: { capture: true },
-      },
-    },
-
-    allFrames: true,
-  },
-
-  PictureInPicture: {
-    parent: {
-      moduleURI: "resource://gre/modules/PictureInPicture.jsm",
-    },
-    child: {
-      moduleURI: "resource://gre/actors/PictureInPictureChild.jsm",
-    },
-
-    allFrames: true,
-  },
-
-  PictureInPictureToggle: {
-    parent: {
-      moduleURI: "resource://gre/modules/PictureInPicture.jsm",
-    },
-    child: {
-      moduleURI: "resource://gre/actors/PictureInPictureChild.jsm",
-      events: {
-        UAWidgetSetupOrChange: {},
-        contextmenu: { capture: true },
-      },
-    },
-
-    allFrames: true,
-  },
-
   PopupBlocking: {
     parent: {
       moduleURI: "resource://gre/actors/PopupBlockingParent.jsm",
