diff --git a/dom/system/NetworkGeolocationProvider.jsm b/dom/system/NetworkGeolocationProvider.jsm
--- a/dom/system/NetworkGeolocationProvider.jsm
+++ b/dom/system/NetworkGeolocationProvider.jsm
@@ -253,28 +253,7 @@ function NetworkGeolocationProvider() {
     isWifiScanningEnabled).  In this case, we use this timer to collect cell/ip
     data and xhr it to the location server.
   */
-  XPCOMUtils.defineLazyPreferenceGetter(
-    this,
-    "_wifiMonitorTimeout",
-    "geo.provider.network.timeToWaitBeforeSending",
-    5000
-  );
 
-  XPCOMUtils.defineLazyPreferenceGetter(
-    this,
-    "_wifiScanningEnabled",
-    "geo.provider.network.scan",
-    true
-  );
-
-  XPCOMUtils.defineLazyPreferenceGetter(
-    this,
-    "_wifiScanningEnabledCountry",
-    "geo.provider-country.network.scan",
-    true
-  );
-
-  this.wifiService = null;
   this.timer = null;
   this.started = false;
 }
@@ -303,8 +282,8 @@ NetworkGeolocationProvider.prototype = {
     this.timer = Cc["@mozilla.org/timer;1"].createInstance(Ci.nsITimer);
     this.timer.initWithCallback(
       this,
-      this._wifiMonitorTimeout,
-      this.timer.TYPE_REPEATING_SLACK
+      0,
+      this.timer.TYPE_ONE_SHOT
     );
   },
 
@@ -315,16 +294,6 @@ NetworkGeolocationProvider.prototype = {
 
     this.started = true;
 
-    if (this.isWifiScanningEnabled) {
-      if (this.wifiService) {
-        this.wifiService.stopWatching(this);
-      }
-      this.wifiService = Cc["@mozilla.org/wifi/monitor;1"].getService(
-        Ci.nsIWifiMonitor
-      );
-      this.wifiService.startWatching(this);
-    }
-
     this.resetTimer();
     LOG("startup called.");
   },
@@ -348,11 +317,6 @@ NetworkGeolocationProvider.prototype = {
       this.timer = null;
     }
 
-    if (this.wifiService) {
-      this.wifiService.stopWatching(this);
-      this.wifiService = null;
-    }
-
     this.listener = null;
     this.started = false;
   },
@@ -419,87 +383,7 @@ NetworkGeolocationProvider.prototype = {
    *                          </code>
    */
   sendLocationRequest(wifiData) {
-    let data = { cellTowers: undefined, wifiAccessPoints: undefined };
-    if (wifiData && wifiData.length >= 2) {
-      data.wifiAccessPoints = wifiData;
-    }
-
-    let useCached = isCachedRequestMoreAccurateThanServerRequest(
-      data.cellTowers,
-      data.wifiAccessPoints
-    );
-
-    LOG("Use request cache:" + useCached + " reason:" + gDebugCacheReasoning);
-
-    if (useCached) {
-      gCachedRequest.location.timestamp = Date.now();
-      if (this.listener) {
-        this.listener.update(gCachedRequest.location);
-      }
-      return;
-    }
-
-    // From here on, do a network geolocation request //
-    let url = Services.urlFormatter.formatURLPref("geo.provider.network.url");
-    LOG("Sending request");
-
-    let xhr = new XMLHttpRequest();
     this.onStatus(false, "xhr-start");
-    try {
-      xhr.open("POST", url, true);
-      xhr.channel.loadFlags = Ci.nsIChannel.LOAD_ANONYMOUS;
-    } catch (e) {
-      this.onStatus(true, "xhr-error");
-      return;
-    }
-    xhr.setRequestHeader("Content-Type", "application/json; charset=UTF-8");
-    xhr.responseType = "json";
-    xhr.mozBackgroundRequest = true;
-    // Allow deprecated HTTP request from SystemPrincipal
-    xhr.channel.loadInfo.allowDeprecatedSystemRequests = true;
-    xhr.timeout = Services.prefs.getIntPref("geo.provider.network.timeout");
-    xhr.ontimeout = () => {
-      LOG("Location request XHR timed out.");
-      this.onStatus(true, "xhr-timeout");
-    };
-    xhr.onerror = () => {
-      this.onStatus(true, "xhr-error");
-    };
-    xhr.onload = () => {
-      LOG(
-        "server returned status: " +
-          xhr.status +
-          " --> " +
-          JSON.stringify(xhr.response)
-      );
-      if (
-        (xhr.channel instanceof Ci.nsIHttpChannel && xhr.status != 200) ||
-        !xhr.response
-      ) {
-        this.onStatus(true, !xhr.response ? "xhr-empty" : "xhr-error");
-        return;
-      }
-
-      let newLocation = new NetworkGeoPositionObject(
-        xhr.response.location.lat,
-        xhr.response.location.lng,
-        xhr.response.accuracy
-      );
-
-      if (this.listener) {
-        this.listener.update(newLocation);
-      }
-
-      gCachedRequest = new CachedRequest(
-        newLocation,
-        data.cellTowers,
-        data.wifiAccessPoints
-      );
-    };
-
-    var requestData = JSON.stringify(data);
-    LOG("sending " + requestData);
-    xhr.send(requestData);
   },
 };
 
